### PR TITLE
Allow browsers to specify fd preopens

### PR DESCRIPTION
**Pull request by chrisdickinson**
A continuation of #59.

Fd preopens are specific to `@bjorn3/browser_wasi_shim`. We expose an option to specify them in `src/interfaces.ts`, but the Fd type is specified as `any[]` because we can't directly reference that package from Deno/Node TS builds.

(This is a v2.0.0 release candidate feature.)

### CODE REVIEW

**Review by zshipko — APPROVED** [review]
Looks good!

**Comment by chrisdickinson** [comment]
I've got a fix for the test failure in #65; a new version of Deno introduced a regression in the build (tracked as https://github.com/denoland/deno/issues/23721.)

**Comment by chrisdickinson** [comment]
@kevinbarabash If you have a moment, could you check this PR out and see if it meets your needs?

**Comment by nilslice** [comment]
@kevinbarabash, if you're unable to verify, then we'll likely merge this if it satisfy's @chrisdickinson's expectations. It's blocking our release though, so we'd like to know if you're interested in confirming. No problem if not!

Let's give this a couple more days, and then proceed as we'd like.

File: src/polyfills/browser-wasi.ts [hunk 1]
```diff
@@ -1,48 +1,29 @@
...
-}
+import { WASI, Fd, File, OpenFile, ConsoleStdout } from '@bjorn3/browser_wasi_shim';
+import { type InternalWasi } from '../interfaces.ts';
 
 export async function loadWasi(
   _allowedPaths: { [from: string]: string },
   enableWasiOutput: boolean,
+  fileDescriptors: Fd[],
 ): Promise<InternalWasi> {
+  console.log('fileDescriptors = ', fileDescriptors);
```
Comment (nilslice) [inline src/polyfills/browser-wasi.ts]: @chrisdickinson - does this `console.log` need to be here, or just leftover from debugging?

**Comment by chrisdickinson** [comment]
I'm going to close this for now – I think we need to revisit how we support WASI in the JS sdk (esp. since Deno dropped support and Bun only has partial support)

**Pull request closed without merging**